src/Mail/Mailable.php

```diff
@@ -32,6 +33,12 @@ public function buildViewData()
     {
         $data = $this->viewData;
 
+        // retrieved saved locale
+        if (isset($data['_saved_locale'])) {
+            App::setLocale($data['_saved_locale']);
+            unset($data['_saved_locale']);
+        }
```

Comment:
I'd recommend also storing the current locale before making the change and restoring it after the mail has been processed just in case.

Reply:
in case of what? can you give more details of why/where you'd want this?

Reply:
As a rough example, lets say that someone was logging messages after the email was sent, if the locale was switched but not changed back then the messages could start appearing in the log in a different locale. Additionally, if multiple mails are processed in a single request (not sure if this applies to queued emails, but this method might be called on regular emails too), a single email could then change the entire application locale for the rest of the request without it being properly reset back to the original locale immediately after sending the email.

Reply:
ok, I'll leave the _current_locale variable in there, it can be used by whoever needs it and shouldn't interfere with other requests.